### DISCUSSION TITLE
Clang lsp doesn't find header

### DISCUSSION THREAD

**Discussion by noahfraiture** [helix-editor/helix#10891] [Q&A] [answered]
Hello, I have some trouble with my C configuration. I learn to work on embedded device with idf.py freeRtos. I thus have an import `#include "freertos/FreeRTOS.h"` that isn't present in my files but will be when I build with the toolchain. IDE like Clion doesn't have a problem like. The consequence is that I have a LOT of diagnostic.

Does anyone know how to solve this issue ?

**Comment by TornaxO7** [comment, marked as answer]
May I ask which build tool you're using?
- `cmake` provides [`CMAKE_EXPORT_COMPILE_COMMANDS`](https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html) which can create a `compile_commands.json` file for `clangd` to resolve the includes.
- For anything else (like `make`), [`bear`](https://github.com/rizsotto/Bear) will be your friend here probably.

**Comment by noahfraiture** [comment]
I think idf use `make`. Thank I will search with the provided link

**Reply by TornaxO7** [reply]
Please mark your question as answered if it helped you
![peepoBearBlush](https://github.com/helix-editor/helix/assets/50843046/84e4dc19-0336-4cd6-8a5c-57d1bbe95d9a)
